### DISCUSSION TITLE
Question/Discussion: about performance with multi core.

### DISCUSSION THREAD

**Discussion by sheeaza** [saghen/blink.cmp#57] [Ideas]
Hi,
I saw that this plugin is with performance in mind and currently using SIMD( low level cpu instruction?) to achieve this.
And another thing I'd like to know is, will this plugin consider taking advantage of multicore to improve the performance?

>The plugin use a 4 stage pipeline: trigger -> sources -> fuzzy -> render

If the `sources` are really huge, I thought using multicore will improve performance a lot in this situation.

### reference
some fuzzy match plugins/tools that use multicore:
- https://github.com/Yggdroot/LeaderF, this fuzzy match plugin use thread pool implemented in C to do match, make it bleeding fast.
- https://github.com/helix-editor/nucleo, fuzzy match engine used by helix, also take advantage of multicore.

Please take this as a discussion not requirement.
Thanks.

**Comment by saghen** [comment]
As is, it can handle 20k+ items in under 4ms on my machine (3700x) which is already a rare case. The typical case falls more into the 0.1-1ms range. Multi core adds higher throughput at the cost of efficiency and complexity which doesn't seem worth it.

WRT further optimization, half the time (or more) is spent in Lua for rendering the windows, so I'd focus my efforts there. I hope that answers!

**Reply by sheeaza** [reply]
Ok, thanks for the answer, so currently there's no such high demand for the multi core.
